gems/aws-sdk-core/lib/aws-sdk-core/plugins/jsonvalue_converter.rb

```diff
@@ -43,15 +43,17 @@ def convert_jsonvalue(m, ref, params, context)
           end
         end
 
-        def serialize_jsonvalue(v, context)
-          unless v.respond_to?(:to_json)
+        def serialize_jsonvalue(value, context)
+          return value if value.is_a?(String)
```

Comment:
Is this correct? json values can be strings however I think it still needs to respond to that. What test does this break?

Reply:
Here's the context: This migrated [test case](https://github.com/smithy-lang/smithy/blob/ac925618457ee9d97da5fd3b63b80e9952ab5cd6/smithy-aws-protocol-tests/model/awsJson1_1/kitchen-sink.smithy#L30) was failing.

Previous to the change above, the output body was escaping twice: `{"JsonValue"=>"\"{\\\"string\\\":\\\"value\\\",\\\"number\\\":1234.5,\\\"boolTrue\\\":true,\\\"boolFa...\\\":false,\\\"array\\\":[1,2,3,4],\\\"object\\\":{\\\"key\\\":\\\"value\\\"},\\\"null\\\":null}\""}`

The first escape happens in the above and second escape during the building the body in the request.

Yeah I could add a bit like, `if value.is_a?(String) && value.responds_to(:to_json)` in the above but shouldn't all ruby strings be able to respond to `to_json`? Lemme know if I'm missing something here.

Reply:
Yeah - I'm going back and forth on this one.   The spec says "When marshaling API requests, SDKs must first encode the shape reference's value into a JSON string. ".  So in this case, if we are given a String, we should probably encode it as a JSON string (which would add extra escaping).

However, to get the test to pass - I think we would need to add special case handling when a test case has input for jsonvalue that we first do a json.parse and then pass that as the input.

Reply:
I asked Kevin for clarification prior to making any changes. He said it seems about right so I'll add in a handling to resolve this test case on our end (and revert the changes I made relating to it)